pytorch_lightning/metrics/functional/classification.py

```diff
@@ -275,6 +275,46 @@ def accuracy(
     return class_reduce(tps, sups, sups, class_reduction=class_reduction)
 
 
+def accuracy_topk(
+        pred: torch.Tensor,
+        target: torch.Tensor,
+        num_classes: Optional[int] = None,
+        topk: int = 1,
+        class_reduction: str = 'micro',
+) -> torch.Tensor:
+    """
+    Computes the topk accuracy classification score
+
+    Args:
+        pred: predicted labels
+        target: ground truth labels
+        num_classes: number of classes
+        class_reduction: method to reduce metric score over labels
+        topk : top kth accuracy required to calculate.
+            - ``'micro'``: calculate metrics globally (default)
+            - ``'macro'``: calculate metrics for each label, and find their unweighted mean.
+            - ``'weighted'``: calculate metrics for each label, and find their weighted mean.
+            - ``'none'``: returns calculated metric per class
+
+    Return:
+         A Tensor with the accuracy score.
+
+    Example:
+
+        >>> x = torch.tensor([0, 1, 2, 3])
+        >>> y = torch.tensor([0, 1, 2, 2])
+        >>> accuracy_topk(x, y)
+        tensor(0.7500)
+    """
+    if not (target > 0).any() and num_classes is None:
+        raise RuntimeError("cannot infer num_classes when target is all zero")
+
+    tps, fps, tns, fns, sups = stat_scores_multiple_classes(
+        pred=pred, target=target, num_classes=num_classes)
+
+    return class_reduce(tps, sups, sups, class_reduction=class_reduction)
```

Comment:
from this implementation it looks like plain accuracy?

Reply:
I have not implemented anything as of now. I was just trying how the API would look.
I have not worked out this PR yet. @rohitgr7 is having a go